src/platforms/ios/iaphandler.mm

```diff
@@ -337,59 +356,96 @@ - (void)request:(SKRequest*)request didFailWithError:(NSError*)error {
 
+  Product* productData = findProduct(productIdentifier);
+  Q_ASSERT(productData);
+
   logger.log() << "Id:" << productIdentifier;
   logger.log() << "Title:" << QString::fromNSString([product localizedTitle]);
   logger.log() << "Description:" << QString::fromNSString([product localizedDescription]);
 
-  NSNumberFormatter* numberFormatter = [[NSNumberFormatter alloc] init];
-  [numberFormatter setFormatterBehavior:NSNumberFormatterBehavior10_4];
-  [numberFormatter setNumberStyle:(NSNumberFormatterStyle)NSNumberFormatterCurrencyStyle];
-  [numberFormatter setLocale:product.priceLocale];
+  QString priceValue;
+  {
+    NSNumberFormatter* numberFormatter = [[NSNumberFormatter alloc] init];
+    [numberFormatter setFormatterBehavior:NSNumberFormatterBehavior10_4];
+    [numberFormatter setNumberStyle:(NSNumberFormatterStyle)NSNumberFormatterCurrencyStyle];
+    [numberFormatter setLocale:product.priceLocale];
 
-  NSString* price = [numberFormatter stringFromNumber:product.price];
-  QString priceValue = QString::fromNSString(price);
+    NSString* price = [numberFormatter stringFromNumber:product.price];
+    priceValue = QString::fromNSString(price);
+    [numberFormatter release];
+  }
 
   logger.log() << "Price:" << priceValue;
-  [numberFormatter release];
 
-  Q_ASSERT(m_pendingProducts.contains(productIdentifier));
-  Product& productData = m_pendingProducts[productIdentifier];
-  productData.m_price = priceValue;
-  productData.m_product = product;
+  QString monthlyPriceValue;
+  NSDecimalNumber* monthlyPriceNS = nullptr;
+  {
+    NSNumberFormatter* numberFormatter = [[NSNumberFormatter alloc] init];
+    [numberFormatter setFormatterBehavior:NSNumberFormatterBehavior10_4];
+    [numberFormatter setNumberStyle:(NSNumberFormatterStyle)NSNumberFormatterCurrencyStyle];
+    [numberFormatter setLocale:product.priceLocale];
+
+    int32_t mounthCount = productTypeToMonthCount(productData->m_type);
+    Q_ASSERT(mounthCount >= 1);
+
+    if (mounthCount == 1) {
+      monthlyPriceNS = product.price;
+    } else {
+      NSDecimalNumber* divider = [[NSDecimalNumber alloc] initWithDouble:(double)mounthCount];
+      monthlyPriceNS = [product.price decimalNumberByDividingBy:divider];
+      [divider release];
+    }
 
-  m_registeredProducts.append(productData);
-  m_pendingProducts.remove(productIdentifier);
+    NSString* price = [numberFormatter stringFromNumber:monthlyPriceNS];
+    monthlyPriceValue = QString::fromNSString(price);
 
-  if (m_pendingProducts.isEmpty()) {
-    logger.log() << "All the products has been registered";
-    m_productsRegistrationState = eRegistered;
-    sortProductsAndCompleteRegistration();
+    [numberFormatter release];
   }
+
+  logger.log() << "Monthly Price:" << monthlyPriceValue;
+
+  productData->m_price = priceValue;
+  productData->m_monthlyPrice = monthlyPriceValue;
+  productData->m_nonLocalizedMonthlyPrice = [monthlyPriceNS doubleValue];
```

Comment:
Isn't `monthlyPriceNS` localized? Why do we call the variable `m_nonLocalized...`. 
Reading in `SKProduct` the `price` property is localized right?

Reply:
`SKProduct` price is an `NSDecimalNumber`. And it's not localized. To localize it we need to run the following code:

```
  NSNumberFormatter* numberFormatter = [[NSNumberFormatter alloc] init];
  [numberFormatter setFormatterBehavior:NSNumberFormatterBehavior10_4];
  [numberFormatter setNumberStyle:(NSNumberFormatterStyle)NSNumberFormatterCurrencyStyle];
  [numberFormatter setLocale:product.priceLocale];

  NSString* price = [numberFormatter stringFromNumber:product.price];
```